presto-main/src/main/java/io/prestosql/metadata/TableLayoutResult.java

```diff
@@ -29,15 +29,22 @@
 
 public class TableLayoutResult
 {
+    private final TableHandle newTableHandle;
```

Comment:
why `newTableHanlde` and not just `tableHandle`?

Reply:
It's just to indicate that this is the new table handle to be used to refer to the result of pushing the predicates into the table scan. It's named that way for clarity, but it doesn't matter in the medium term since I'm planning to remove this class altogether once the replacement from https://github.com/prestosql/presto/wiki/Pushdown-of-complex-operations is in place and we give connectors some time to migrate.